### DISCUSSION TITLE
In "connect": How to reference ID of the document that is being created?

### DISCUSSION THREAD

**Discussion by tomaszferens** [prisma/prisma#6619] [Q&A] [answered]
My schema looks more or less like this:

```
model Supplier {
  id        String     @id @default(cuid())
  customers Customer[]
  orders    Order[]
}

model Customer {
  id        String     @id @default(cuid())
  suppliers Supplier[]
  orders    Order[]
}

model Order {
  id         String   @id @default(cuid())
  customerId String
  customer   Customer @relation(fields: [customerId], references: [id])
  supplierId String
  supplier   Supplier @relation(fields: [supplierId], references: [id])
}
```

I want to create in a single transaction a supplier with a customer and an order.

My attempt:

```
await prisma.supplier.create({
    data: {
        customers: {
            create: {
                orders: {
                    create: {
                        supplier: {
                            connect: {
                                id: '??SUPLIER_ID??',
                            },
                        },
                        customer: {
                            connect: {
                                id: '??CUSTOMER_ID??',
                            },
                        },
                    },
                },
            },
        },
    },
});
```

And I'm stuck here, not sure how to get a supplierId and customerId when creating an order... any ideas? :)

**Comment by tomaszferens** [comment, marked as answer]
Okay I figured out the query can be rewritten to:
```
await prisma.supplier.create({
    data: {
        orders: {
            create: {
                customer: {
                    create: {},
                },
            },
        },
    },
});
```

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
